paasta_tools/paasta_maintenance.py

```diff
@@ -0,0 +1,276 @@
+# limitations under the License.
+import argparse
+import datetime
+import json
+from socket import gethostbyname
+
+from dateutil import tz
+from pytimeparse import timeparse
+from requests import Request
+from requests import Session
+
+from paasta_tools.mesos_tools import get_mesos_leader
+from paasta_tools.mesos_tools import MesosMasterConnectionError
+
+
+PORT = 5050
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        '-d', '--duration',
+        type=timedelta_type,
+        default='1h',
+        help="Duration of the maintenance window. Any pytimeparse unit is supported.",
+    )
+    parser.add_argument(
+        '-s', '--start',
+        default=now(),
+        help="Time to start the maintenance window. Defaults to now.",
+    )
+    parser.add_argument(
+        'action',
+        choices=['drain', 'undrain', 'down', 'up', 'status'],
+        help="Action to perform on the speicifed hosts",
+    )
+    parser.add_argument(
+        'hostname',
+        nargs='*',
+        help="Hostname(s) of machine(s) to start draining.",
+    )
+    options = parser.parse_args()
+    return options
+
+
+def base_api():
+    leader = get_mesos_leader()
+
+    def execute_request(method, endpoint, **kwargs):
+        url = "http://%s:%d/%s" % (leader, PORT, endpoint)
+        timeout = 15
+        s = Session()
+        req = Request(method, url, **kwargs)
+        prepared = s.prepare_request(req)
+        try:
+            resp = s.send(prepared,
+                          timeout=timeout,
+                          auth=load_credentials(),
+                          )
+            resp.raise_for_status()
+        except Exception as e:
+            print("Error executing API request calling %s. Got response code %d with error %s" %
+                  (endpoint, resp.status_code, e))
+    return execute_request
+
+
+def master_api():
+    def execute_master_api_request(method, endpoint, **kwargs):
+        base_api_client = base_api()
+        return base_api_client(method, "master/%s" % endpoint, **kwargs)
+    return execute_master_api_request
+
+
+def maintenance_api():
+    def execute_schedule_api_request(method, endpoint, **kwargs):
+        master_api_client = master_api()
+        return master_api_client(method, "maintenance/%s" % endpoint, **kwargs)
+    return execute_schedule_api_request
+
+
+def get_schedule_client():
+    def execute_schedule_api_request(method, endpoint, **kwargs):
+        maintenance_api_client = maintenance_api()
+        return maintenance_api_client(method, "schedule/%s" % endpoint, **kwargs)
+    return execute_schedule_api_request
+
+
+def get_maintenance_schedule():
+    client_fn = get_schedule_client()
+    return client_fn(method="GET", endpoint="")
+
+
+def get_maintenance_status():
+    client_fn = get_schedule_client()
+    return client_fn(method="GET", endpoint="status")
+
+
+def timedelta_type(value):
```

Comment:
ah right, ok. heh, TIL!

ok, in that case... can you be more explicit? call it ``parse_timedelta`` or something. Also, you should probably catch the exception that'll be thrown by ``timeparse.timeparse(value)`` when this isn't legit and throw the correct ``argparse.ArgumentTypeError`` so that the output is formatted correctly by argparse